### ISSUE TITLE
[A11y] Unify Talkback and voice over Textfield announcements

### ISSUE THREAD

**Issue by chunhtai** [flutter/flutter#183808]
If you have a textfield with semantics label and hint and screen reader focuses it, the talkback will announce them after the textfield while iOS will announce label, textfield, hint.

For example:
```dart
import 'package:flutter/material.dart';

/// Flutter code sample for [TextField].

class TextFieldExampleApp extends StatelessWidget {
  const TextFieldExampleApp({super.key});
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        appBar: AppBar(title: const Text('Obscured Textfield')),
        body: Center(
          child: SizedBox(
            width: 250,
            child: Semantics(
              label: 'label',
              hint: 'hint',
              child: TextField(controller: TextEditingController(text: '123')),
            ),
          ),
        ),
      ),
    );
  }
}

void main() => runApp(const TextFieldExampleApp());
```

in talkback it annonces: 'editing, 123, edit text, label, hint'
in voiceover it annonces: 'label, textfield, is editing, 123, <a long pause>, hint'

This is due to how we fed this string into the accessibility API

in talkback, we combine both label and hint into one string and fed into ANI.setHintText which will be append at the end of the announcement. We could uses ANI.setContentDescription to make the string to be pronounced first. This would however override announcement of the "editing, 123", but that is something we could append to the semantics label from the framework side.

In voiceover, label is fed to accessibilityLabel and hint to accessibilityHint, where accessibilityLabel will be announced first, and accessibilityHint will be announced last.

Either way, we should have control over the order they are pronounced.

**Comment by zywek123** [comment]
Whenever a developer can have a choice it's good, but it's not good to impose the one and only right way to do things.
Because in that case, let's try to think about unifying it for the other platforms, that is, for windows and Linux, because on Osx it is similar to iOS.
Surely users won't mind and won't wonder why in one app it tells them something this way and in another it tells them something different, and I say this as a blind user and app developer.
